p2p/src/network/pubsub/p2p_network_pubsub_reducer.rs

```diff
@@ -173,22 +181,31 @@ impl P2pNetworkPubsubState {
                 message,
                 seen_limit,
             } => {
+                // Check that if we can extract source from message, this is pre check
+                if source_from_message(&message).is_err() {
+                    let dispatcher = state_context.into_dispatcher();
+                    dispatcher.push(P2pDisconnectionAction::Init {
```

Comment:
Shouldn't here the message be rejected instead? and that will cause the peer to be disconnected